lib/Onyx.js

```diff
@@ -359,8 +359,9 @@ function keysChanged(collectionKey, partialCollection) {
  * @private
  * @param {String} key
  * @param {*} data
+ * @param {*} previousData
  */
-function keyChanged(key, data) {
+function keyChanged(key, data, previousData) {
```

Comment:
Yeah, it's because if we try to access the cache inside of `keyChanged()`, the cache has already been updated with the new value. We have to get the previous data _before_ it's set into the cache, and then pass it down.

Reply:
Got it. Ok, I'm having a few thoughts about this approach...

- React components already store their values in `prevState` and they are accessible in the `setState()` callback. If we want to bail out of the `setState()` call we just return `null` so I think this could be a good place to put the logic for `withOnyx()` subscribers
- For the `Onyx.connect()` case we can't see the last values that were last passed to callback so maybe an alternative could be to have something like a `lastCalledWith` value for `Onyx.connect()` that we can compare against
- I like these alternatives better since we will not have to store a temp copy of the state anytime we are calling `keyChanged()` and we will be comparing not what is in the cache but what the component/subscriber has

Reply:
OK, I implemented the `prevState` that you mentioned for `withOnyx()` and it works well. I will try out the `lastCalledWith` and see if I can figure it out.